synfig-core/src/synfig/string_helper.cpp

```diff
@@ -0,0 +1,104 @@
+/* === S Y N F I G ========================================================= */
+/*!	\file string_helper.cpp
+**	\brief Implementation of helper functions to handle strings
+**
+**	$Id$
+**
+**	\legal
+**	Copyright (c) 2002-2005 Robert B. Quattlebaum Jr., Adrian Bentley
+**	Copyright (c) 2021 Rodolfo R Gomes
+**
+**	This package is free software; you can redistribute it and/or
+**	modify it under the terms of the GNU General Public License as
+**	published by the Free Software Foundation; either version 2 of
+**	the License, or (at your option) any later version.
+**
+**	This package is distributed in the hope that it will be useful,
+**	but WITHOUT ANY WARRANTY; without even the implied warranty of
+**	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+**	General Public License for more details.
+**	\endlegal
+*/
+
+/* ========================================================================= */
+
+#ifdef USING_PCH
+#	include "pch.h"
+#else
+# ifdef HAVE_CONFIG_H
+#  include <config.h>
+# endif
+
+#include <synfig/string_helper.h>
+
+#include <algorithm>
+#include <locale>
+#endif
+
+std::string
+synfig::remove_trailing_zeroes(std::string text)
+{
+	std::locale l(setlocale(LC_NUMERIC, nullptr));
+	const char decimal_point = std::use_facet< std::numpunct<char> >(l).decimal_point();
+	const size_t decimal_point_pos = text.find(decimal_point);
+
+	if (decimal_point_pos == text.npos) {
+		text += decimal_point + '0';
+	} else {
+		const size_t last_non_zero_pos = text.find_last_not_of('0');
+		text = text.substr(0, std::max(decimal_point_pos, last_non_zero_pos) + 1);
+	}
+	return text;
+}
+
+std::string
+synfig::trim(std::string text)
+{
+	return right_trim(left_trim(text));
+}
+
+std::wstring
+synfig::trim(std::wstring text)
+{
+	return right_trim(left_trim(text));
+}
+
+std::string
+synfig::left_trim(std::string text)
+{
+	text.erase(text.begin(),
+			   std::find_if(text.begin(), text.end(),
+							[](int chr) { return !std::isspace(chr);})
+			   );
+	return text;
+}
+
+std::wstring
+synfig::left_trim(std::wstring text)
+{
+	text.erase(text.begin(),
+			   std::find_if(text.begin(), text.end(),
+							[](int chr) { return !std::isspace(chr);})
+			   );
+	return text;
+}
+
+std::string
+synfig::right_trim(std::string text)
```

Comment:
I suggest use `const std::string& text` if you want non-destructive trimming or `std::string&& text` for destructive.
Just to understand what's happening with the original string.

Reply:
As with C++11, it would automatically use `move`, I thought it wouldn't be necessary.
But I'll change it.

Reply:
Done. Changed all function signatures to `const std::(w)string&`.